editor/editor_dock_manager.h

```diff
@@ -97,7 +124,7 @@ class EditorDockManager : public Object {
 	Vector<DockSplitContainer *> hsplits;
 
 	Vector<WindowWrapper *> dock_windows;
-	TabContainer *dock_slot[DOCK_SLOT_MAX];
+	EditorDockTabContainer *dock_slot[DOCK_SLOT_MAX];
```

Comment:
https://github.com/godotengine/godot/pull/106762/files#r2110295167

EditorDockTabContainer's logic is entirely internal, so you can use TabContainer.